messaging/usecase/room_usecase.go

```diff
@@ -3,109 +3,101 @@ package usecase
 import (
 	"chat/domain"
 	"chat/utils/errors"
+	"context"
 	"fmt"
-	"log"
 	_ "reflect"
 	"time"
 )
+
 /*
 
 	SaveRoom( room *ChatRoom) error
-	GetChatRoomsFor( studentID string) (*StudentChatRooms, error)
-	EditChatRoomParticipants( room *ChatRoom) error
+	AddUserToRoom(roomID string, userID string) error
+	RemoveUserFromRoom(roomID string, userID string) error
+	GetChatRoomsFor( userID string) (*StudentChatRooms, error)
 	DeleteRoom( userID string, roomID string) error
 */
 type roomUseCase struct {
 	rr      domain.RoomRepository
-	st 		domain.StudentRepository
+	sr      domain.StudentRepository
 	timeout time.Duration
 }
 
-func (u *roomUseCase) SaveRoom(room *domain.ChatRoom) error {
+func NewRoomUseCase(rr domain.RoomRepository, sr domain.StudentRepository, t time.Duration) domain.RoomUseCase {
+	return &roomUseCase{rr: rr, sr: sr, timeout: t}
+}
+
+// SaveRoom should add room to chat.room & chat.student_rooms for all participants
+func (u *roomUseCase) SaveRoom(ctx context.Context, room *domain.ChatRoom) error {
 
-	_,err := u.rr.GetRoom(room.RoomID)
-	if err==nil {
+	_, err := u.rr.GetRoom(ctx, room.RoomID)
+	if err == nil {
 		return errors.NewConflictError(fmt.Sprintf("Room with ID %s already exists", room.RoomID))
 	}
-	_,err=u.st.GetStudent(room.Admin.ID)
-	if err==nil {
-		return errors.NewConflictError(fmt.Sprintf("The admin with ID  %s does not exist", room.Admin.ID))
-	}
-	err = u.rr.SaveRoom( room )
-	if err!=nil {
-		return err
+
+	for _, participant := range room.Students {
+		_, err = u.sr.GetStudent(ctx, participant.ID)
+		if err != nil {
+			return errors.NewConflictError(fmt.Sprintf("The participant with ID %s does not exist", participant.ID))
+		}
 	}
 
-   return err
+	return u.rr.SaveRoomAndAddRoomForAllParticipants(ctx, room)
+}
+
+// AddUserToRoom should add user to room in chat.room and add room to student in chat.student_rooms
+func (u *roomUseCase) AddUserToRoom(ctx context.Context, roomID string, userID string) error {
+	return u.rr.AddParticipantToRoomAndAddRoomForParticipant(ctx, roomID, userID)
 }
 
-func (u *roomUseCase)  GetStudentChatRoomsFor( studentID string) (*domain.StudentChatRooms, error){
+// RemoveUserFromRoom should remove user from room in chat.room and remove room from user in chat.student_rooms
+func (u *roomUseCase) RemoveUserFromRoom(ctx context.Context, roomID string, userID string) error {
+	return u.rr.RemoveParticipantFromRoomAndRemoveRoomForParticipant(ctx, roomID, userID)
```

Comment:
all the contexts need to with timeout like [here](https://github.com/Airbenders-490/profile/blob/ff69d292339b2ac39886e29d04847dd72e269bd4/Review/usecase/review_usecase.go#L31)